### PR TITLE
Added test scan routes api 

### PR DESCRIPTION
**Pull request by tushar-balwani**
# Description

Added **`test scan route`** api in tio scans.
```
scan_routes_info = tio.scans.check_auto_targets(10, 5, targets=['127.0.0.1'])
```

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Added below tests for **check_auto_targets** api
- [x] test_scan_check_auto_targets_success
- [x] test_scan_check_auto_targets_limit_typeerror
- [x] test_scan_check_auto_targets_matched_resource_limit_typeerror
- [x] test_scan_check_auto_targets_network_uuid_unexpectedvalueerror
- [x] test_scan_check_auto_targets_network_uuid_typeerror
- [x] test_scan_check_auto_targets_tags_unexpectedvalueerror
- [x] test_scan_check_auto_targets_tags_typeerror
- [x] test_scan_check_auto_targets_targets_typeerror

**Test Configuration**:
* Python Version(s) Tested: 3.8.6
* Tenable.sc version (if necessary):

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tenable/io/scans.py [hunk 1]
```diff
@@ -103,7 +107,7 @@ def _create_scan_document(self, kw):
                 default='basic',
                 choices=list(templates.keys())
             )]
-            del(kw['template'])
+            del kw['template']
```
Comment (mvalliammal) [inline tenable/io/scans.py:110]: Why the name is kw?. why cant it be more meaningful?

**Review by mvalliammal — COMMENTED** [review]
Line number 663, status variable is used for assignment, it had not been reused

File: tenable/io/scans.py (hunk 1 above)
Comment (tushar-balwani) [reply tenable/io/scans.py]: updated variable name

**Review by valliammal — APPROVED** [review]
Approved the changes

**Comment by levwais** [comment]
please resolve merge conflicts

**Review by levwais — APPROVED** [review]

**Pull request merged**